core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -229,6 +229,8 @@ class KafkaApis(val requestChannel: RequestChannel,
         case ApiKeys.ENVELOPE => handleEnvelope(request, requestLocal)
         case ApiKeys.DESCRIBE_CLUSTER => handleDescribeCluster(request)
         case ApiKeys.DESCRIBE_PRODUCERS => handleDescribeProducersRequest(request)
+        case ApiKeys.UNREGISTER_BROKER => maybeForwardToController(request,
```

Comment:
nit: we already have a `forwardToControllerOrFail` method which can be used here.

Reply:
+1

Reply:
good idea